workitem/workitem_repository.go

```diff
@@ -193,12 +192,12 @@ func (r *GormWorkItemRepository) FindSecondItem(order *float64, secondItemDirect
 	Item := WorkItem{}
 	var tx *gorm.DB
 	switch secondItemDirection {
-	case DirectionBelow:
-		// Finds the item below which reorder item has to be placed
-		tx = r.db.Where("execution_order < ?", order).Order("execution_order desc", true).Last(&Item)
-
 	case DirectionAbove:
 		// Finds the item above which reorder item has to be placed
+		tx = r.db.Where("execution_order < ?", order).Order("execution_order desc", true).Last(&Item)
```

Comment:
@DhritiShikhar as discussed on BJNs: I made a mistake (`max(id)`) that you fixed (`max(execution_order)`) but then you compared it against an `id` field which is wrong.